src/property.ts

```diff
@@ -215,27 +215,22 @@ export class PropertiesManager {
   }
 
   set(propertiesToSet: Properties): void {
-    this.properties = {
-      ...Object.keys(propertiesToSet).map((propertyName) => [
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
     Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
+      if (!((propertyName as KnownProperty) in this.properties)) {
```

Comment:
I prefer o[k] === undefined for this pattern, and I think that's what we use elsewhere.